### PR TITLE
Invoke

### PR DESCRIPTION
**Pull request by ivoflipse**
Tried making a recipe for invoke

### CODE REVIEW

**Comment by ivoflipse** [comment]
I actually wanted to make sure the test suite was run as well, but looking at the .travis.yml file, the only test I could find was: `inv coverage --no-html`

I couldn't find this command in the repo, because apparently it was in a separate repo: 
https://github.com/pyinvoke/invocations/blob/master/invocations/testing.py

So I considered making a package for **invocations** but it relies on invoke itself. So not really sure how to deal with that. Any suggestions?

**Comment by jakirkham** [comment]
I wouldn't sweat running the test suite here. This is a pure Python package that undergoes CI and has proper releases. If we just verify that it exists by importing and running help, then we should be fine.

File: recipes/invoke/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{% set version = "0.12.2" %}
+
+package:
+  name: invoke
+  version: {{ version }}
+
+source:  
+  git_url: https://github.com/pyinvoke/invoke.git
+  git_tag:  {{ version }}
```
Comment (jakirkham) [inline recipes/invoke/meta.yaml]: Could we use the PyPI release here?

File: recipes/invoke/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - invoke.vendor.yaml2    # [py2k]
+    - invoke.vendor.yaml3    # [py3k]
+
+  commands:
+    - invoke --help
+    - inv --help
+
+about:
+  home: http://docs.pyinvoke.org
+  license: BSD License
```
Comment (jakirkham) [inline recipes/invoke/meta.yaml]: What version of BSD? Also, adding "License" is a bit redundant and can be dropped.

Comment (ivoflipse) [inline recipes/invoke/meta.yaml]: I think I just copied it from Pypi, which isn't more specific, nor is his setup.py

File: recipes/invoke/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - invoke.vendor.yaml2    # [py2k]
+    - invoke.vendor.yaml3    # [py3k]
+
+  commands:
+    - invoke --help
+    - inv --help
+
+about:
+  home: http://docs.pyinvoke.org
+  license: BSD
```
Comment (ocefpaf) [inline recipes/invoke/meta.yaml]: Do you known what version of the BSD license is used?

**Comment by ocefpaf** [comment]
> I wouldn't sweat running the test suite here. This is a pure Python package that undergoes CI and has proper releases. If we just verify that it exists by importing and running help, then we should be fine.

:+1: to that.

I have just a quick question regarding the license and we are ready to merge.

File: recipes/invoke/meta.yaml (hunk 3 above)
Comment (ivoflipse) [inline recipes/invoke/meta.yaml]: His [setup.py](https://github.com/pyinvoke/invoke/blob/0f90fa5f6e35a74ff5cf364111a1663e491ba860/setup.py) only mentions `'License :: OSI Approved :: BSD License',` and `license='BSD',` 

Here's a [link](https://github.com/pyinvoke/invoke/blob/0f90fa5f6e35a74ff5cf364111a1663e491ba860/LICENSE) to his actual license

So after checking Wikipedia, I guess it's the 2-clause license version

Comment (ocefpaf) [inline recipes/invoke/meta.yaml]: People have been copying and pasting that PyPI classifier (`'License :: OSI Approved :: BSD License',` and `license='BSD',` ) for so long and never realized it is not helpful.

I agree. It is `BSD 2-Clause`. Please update the recipe to reflect that.

**Comment by ivoflipse** [comment]
@ocefpaf and @jakirkham I've adjusted the license info and switched to PyPi, anything else?

File: recipes/invoke/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  version: {{ version }}
+
+source:  
+  fn: invoke-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/source/i/invoke/invoke-{{ version }}.tar.gz
+  md5: f46192ea664325464a02f1c7f735e3cb
+
+build:
+  number: 0
+  script: python setup.py install
```
Comment (pelson) [inline recipes/invoke/meta.yaml]: It is unusual to use setuptools and not need to ```python setup.py install --single-version-externally-managed --record=record.txt```. Another option is to depend on pip and ```pip install --no-deps .```.

**Comment by jakirkham** [comment]
Basically just @pelson's comment above.

File: recipes/invoke/meta.yaml (hunk 4 above)
Comment (ivoflipse) [inline recipes/invoke/meta.yaml]: Right, I thought this script element was just a shortcut for not needing the build.bat and build.sh files. So you want me to change the script line into: `python setup.py install --single-version-externally-managed --record=record.txt` then?

Comment (jakirkham) [inline recipes/invoke/meta.yaml]: Yes please.

Comment (jakirkham) [inline recipes/invoke/meta.yaml]: There is no need for `build.bat` and `build.sh`. Just the options here will be fine.

Comment (ivoflipse) [inline recipes/invoke/meta.yaml]: Fixed @jakirkham and @pelson

File: recipes/invoke/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+about:
+  home: http://docs.pyinvoke.org
+  license: BSD-2-Clause
+  summary: 'Pythonic task execution'
+
+extra:
+  recipe-maintainers:
+    - ivoflipse
+    - Korijn
```
Comment (jakirkham) [inline recipes/invoke/meta.yaml]: @Korijn, are you happy being added as a maintainer here?

Comment (Korijn) [inline recipes/invoke/meta.yaml]: Very happy! Thanks for putting so much effort into conda-forge. We've been having a hard time submitting our recipes in the past, this is brilliant!

Comment (jakirkham) [inline recipes/invoke/meta.yaml]: Well @pelson is the mastermind, might want to thank him too. 😉

**Comment by jakirkham** [comment]
LGTM. Thanks everyone.

**Pull request merged**